libraries/adaptive-expressions/src/builtinFunctions/dateReadBack.ts

```diff
@@ -36,9 +36,9 @@ export class DateReadBack extends ExpressionEvaluator {
             const dateFormat = 'YYYY-MM-DD';
             ({ value, error } = InternalFunctionUtils.parseTimestamp(args[0]));
             if (!error) {
-                const timestamp1: Date = new Date(value.format(dateFormat));
+                const timestamp1: Date = new Date(moment(value).format(dateFormat));
                 ({ value, error } = InternalFunctionUtils.parseTimestamp(args[1]));
-                const timestamp2: string = value.format(dateFormat);
+                const timestamp2: string = moment(value).format(dateFormat);
```

Comment:
No need to include `: string` that will be properly inferred. Same with `: Date` above and `: TimexProperty` below.